ledger/src/blockstore.rs

```diff
@@ -1051,6 +1051,26 @@ impl Blockstore {
             });
 
             if erasure_config != erasure_meta.config {
+                let coding_start = erasure_meta.first_coding_index;
+                let coding_end = coding_start + erasure_meta.size as u64;
+                let mut conflicting_shred = None;
+                for coding_index in coding_start..coding_end {
+                    if let Ok(Some(shred)) = self.get_coding_shred(slot, coding_index) {
+                        conflicting_shred = Some(shred);
```

Comment:
should we add a sanity check here that this `conflicting_shred` has a different `num_data_shreds()` or `num_coding_shreds` than the newly received `shred`?